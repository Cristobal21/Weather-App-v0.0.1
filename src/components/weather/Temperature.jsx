



export const Temperature = ({ main }) => {

    const { temp, min, max } = main


    return (
        <div>
            <h4>Temperatura</h4>
            <p>{`Temperatura actual: ${ temp }`}</p>
            <p>{`Temperatura mínima: ${ min }`}</p>
            <p>{`Temperatura máxima: ${ max }`}</p>
        </div>
    )
}
