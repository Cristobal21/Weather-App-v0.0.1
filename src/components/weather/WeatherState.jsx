
export const WeatherState = ({ main, wind, weather }) => {

    const { velocidad } = wind
    const { humedad } = main
    const { estado } = weather


    return (
        <div>
            <h4>Precipitaciones</h4>
            <p>{`Humedad: ${ humedad }`}</p>
            <p>{`Velocidad viento: ${ velocidad }`}</p>
            <p>{`Estado actual: ${ estado }`}</p>
        </div>
    )
}
