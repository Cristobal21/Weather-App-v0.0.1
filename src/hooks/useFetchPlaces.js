import { useEffect, useState } from "react"
import { getData } from "../helpers/getData"

export const useFetchPlaces = ( lugar ) => {

    const [lugares, setLugares] = useState([])

    const getPlaces = async () => {
        const newPlaces = await getData( lugar )
        setLugares( newPlaces )
    }

    useEffect(() => {

        getPlaces()

    }, [])


    return {
        lugares
    }
}
