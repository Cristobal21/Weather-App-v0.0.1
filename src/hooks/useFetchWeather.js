import { useEffect, useState } from "react"
import { getClima } from "../helpers/getClima"

export const useFetchWeather = ( lugar ) => {

    const [clima, setClima] = useState([])


    const getWeather = async () => {
        const newWeather = await getClima( lugar )
        setClima(newWeather)
    }
    
    
    useEffect(() => {
        getWeather()
    }, [])
    
    return {
        clima
    }
}
