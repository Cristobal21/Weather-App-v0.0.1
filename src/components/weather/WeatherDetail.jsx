
import { Tabs, Tab } from 'react-bootstrap'
import { Temperature } from './Temperature'
import { WeatherState } from './WeatherState'

// export const WeatherDetail = ({ id, main, name, sys, visibility, wind, weather }) => {
export const WeatherDetail = () => {

    // const { lat, lon } = coord // -> Se recibe por paramaetros
    // const { humedad, temp, min, max } = main
    // const { pais } = sys
    // const { velocidad } = wind
    // const { estado } = weather

    return (
        <div>

            <h1>Weather Detail</h1>
            <hr />

            <Tabs
                defaultActiveKey="temp"
                id="example"
                className='mb-3'
                justify
            >
                <Tab eventKey="temp" title="Temperatura">
                    <Temperature/>
                </Tab>
                <Tab eventKey="precip" title="Precipitaciones">
                    <WeatherState/>
                </Tab>
            </Tabs>



            

            

            

            {/* <h4>Coordenadas</h4>
            <p>{`Latitud: ${lat}`}</p>
            <p>{`Longitud: ${lon}`}</p> */}

            

            {/* <div>
            {
                clima.map( (data) => (
                    <PlaceItem
                        key={ data.id }
                        { ...data }
                    />
                ) )
            }
            </div> */}
        </div>
    )
}
