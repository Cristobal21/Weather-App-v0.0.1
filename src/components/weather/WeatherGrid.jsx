import { useFetchWeather } from "../../hooks/useFetchWeather"
import { WeatherDetail } from "./WeatherDetail"
import PropTypes from 'prop-types'




export const WeatherGrid = ({ lugar }) => {

    const { clima } = useFetchWeather(lugar)

    return (
        <>
            <p className="mt-3">{`Clima actual en: ${ lugar }`}</p>
            <hr />

            <div>
                {
                    clima.map( (data) => (
                        <WeatherDetail
                            // key={ data.id }
                            // { ...data }
                        />
                    ) )
                }
            </div>
        </>
    )
}


WeatherGrid.propTypes = {
    lugar: PropTypes.string.isRequired
}