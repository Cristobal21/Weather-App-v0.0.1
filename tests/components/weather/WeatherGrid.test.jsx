import { render, screen } from "@testing-library/react"
import { WeatherGrid } from "../../../src/components/weather/WeatherGrid"
import { useFetchWeather } from "../../../src/hooks/useFetchWeather"

jest.mock('../../../src/hooks/useFetchWeather')

describe('Pruebas en <WeatherGrid/>', () => {

    const lugar = 'Miami'

    test('Debe hacer match con el snapshot', () => {

        useFetchWeather.mockReturnValue({
            clima: []
        })

        const { container } = render(<WeatherGrid lugar={lugar} />)
        expect(container).toMatchSnapshot()
    })

    // test('Debe cargar la info cuando se llame al useFetchWeather', () => {

    //     const weather = [
    //         {
    //             id: 'abc',
    //             name: 'Buin, Region Metropolitana',
    //             visibility: 9000,
    //             coord: {
    //                 lon: 1234,
    //                 lat: -1234,
    //             },
    //             main: {
    //                 humedad: 34,
    //                 temp: 13.11,
    //                 min: 7,
    //                 max: 15
    //             },
    //             sys: {
    //                 pais: 'Chile',
    //                 id: 'CL'
    //             },
    //             wind: {
    //                 velocidad: '1.4km/h'
    //             },
    //             weather: {
    //                 estado: 'Nublado'
    //             },
    //         }
    //     ]

    //     useFetchWeather.mockReturnValue({
    //         clima: weather
    //     })

    //     render(<WeatherGrid lugar={ lugar }/>)

    // })
})