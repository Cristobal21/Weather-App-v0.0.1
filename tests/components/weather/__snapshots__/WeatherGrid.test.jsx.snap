// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <WeatherGrid/> Debe hacer match con el snapshot 1`] = `
<div>
  <p
    class="mt-3"
  >
    Clima actual en: Miami
  </p>
  <hr />
  <div />
</div>
`;
